### PR TITLE
Styles refactor according to stylelint

### PR DESCRIPTION
**Pull request by lukasaric**
This PR refactors vue components on client according to `stylelint` rules.

### CODE REVIEW

File: client/components/course/Revisions/EntitySidebar.vue [hunk 1]
```diff
@@ -56,6 +56,11 @@ export default {
 <style lang="scss" scoped>
 $revision-padding: 32px;
 
+.selected-revision {
+  color: #fff !important;
```
Comment (abasic) [inline client/components/course/Revisions/EntitySidebar.vue]: Can this be done with the `!import` rule?

Comment (lukasaric) [reply client/components/course/Revisions/EntitySidebar.vue]: You mean without `!important` rule? I tried, but main selector `.revision` has color on it's own so when I extend this class on `.selected.revision:hover`, it overrides color applied on this selector.

Comment (abasic) [reply client/components/course/Revisions/EntitySidebar.vue]: The `!important` rule can be avoided in two ways:
1) We use a `mixin` instead of an extended `class`.
2) we continue to use `@extend` on the `.selected-revision` `class` but we move the definition of the `class` below the definition of the `.revision` `class`.

The reason why both would work is the cascading rule of `CSS`. Currently, it does not work because the definition of the `.selected-revision` `class` is above the `.revision`. The way `class` extension works is like this:
 - it takes the names of the `classes` that are extending a `class` and add them to the definition of the rules for the extended `class`. Example:
```
 .a {
   margin: 0;
 }
 
 .b {
   margin: 10px;
 }
 
 .c {
   @extend .a;
   
   color: white;
 }
```
 
 becomes:
 
```
 .a, .c {
   margin: 0;
 }
 
 .b {
   margin: 10px;
 }
 
 .c {
   color: white;
 }
```
 
So, even though in the `SCSS` the definition of the `.c` `class` is below the `.b` `class`. in the final version of the `CSS` the margin rule for the `.c` is above the one for the `.b`. In this case, if there were a tag that had both of these `classes`, the margin rule of the `.b` `class` would be used. The reason why a `mixin` would work is that the `mixins` code is inserted into the `class` definitions or moving the `class` definition below would work is because the extension happens after the `.revision` `class` definition.

tl;dr Use a `mixin` since you are not using the `.selected-revision` `class` in the `HTML`.

File: client/assets/stylesheets/common/_variables.scss [hunk 2]
```diff
@@ -4,3 +4,6 @@ $font-weight-regular: 400;
 $font-weight-medium: 500;
 $font-weight-semibold: 600;
 $font-weight-bold: 700;
+
+// Font family
+$font-family-icons: "Material Design Icons";
```
Comment (kronicker) [inline client/assets/stylesheets/common/_variables.scss]: Technically we could end up using multiple icon fonts so I would rename it to `font-family-icons`

Comment (kronicker) [reply client/assets/stylesheets/common/_variables.scss]: Since you extracted icons font to a variable you can do the same with others.
As far as I see we have two options:
1. `Poppins, Helvetica, Arial, sans-serif` as the primary font and
2. `Roboto, Helvetica, Arial, sans-serif` as secondary.

File: client/components/course/Revisions/EntitySidebar.vue [hunk 3]
```diff
@@ -56,6 +56,11 @@ export default {
 <style lang="scss" scoped>
 $revision-padding: 32px;
 
+@mixin selected-revision() {
```
Comment (kronicker) [inline client/components/course/Revisions/EntitySidebar.vue]: Parens are redundant when you have no args. Also on `@include` statements ⬇️  as well.

File: poi.config.js [hunk 4]
```diff
@@ -64,6 +64,12 @@ module.exports = {
         files: ['.brandrc', '.brandrc.js'],
         imagesPath
       }
+    },
+    {
```
Comment (kronicker) [inline poi.config.js]: Move opening bracket to the line above. Check previous occurrences and be consistent 💅

File: client/components/content-elements/tce-modal/edit/Modal.vue [hunk 5]
```diff
@@ -97,15 +98,15 @@ function toggleFocusTrap(focusTrap, condition) {
   max-width: 640px;
   margin: 0 auto;
   padding: 4px;
-  font: 14px Roboto, Helvetica, Arial;
+  font: 14px $font-family-secondary;
   text-align: left;
   border: none;
   border-radius: 2px;
   box-shadow:
     0 11px 15px -7px rgba(0,0,0,0.2),
```
Comment (abasic) [inline client/components/content-elements/tce-modal/edit/Modal.vue]: Missing spaces in all of the `rgba` rules.

File: client/components/common/Meta/Switch.vue [hunk 6]
```diff
@@ -132,13 +115,30 @@ label {
...
-  .checked & {
+    &::before, &::after {
+      left: 18px;
+    }
+  }
+
+  .lever:active::before {
     transform: scale(2.4);
-    background-color: transparentize($checked, 0.85);
+    background-color: rgba(0,0,0,0.08);
```
Comment (abasic) [inline client/components/common/Meta/Switch.vue]: Missing spaces in `rgba` rule.
